### ISSUE TITLE
Allow for smaller minimum width

### ISSUE THREAD

**Issue by frandavid100** [toolstack/Folio#37]
Notes' width doesn't seem to adjust to the window:

![Captura desde 2024-03-07 22-50-53](https://github.com/toolstack/Folio/assets/694794/dd2126f4-fca9-49ea-b47a-1155a4a3ab8f)

Is this a bug?

**Comment by toolstack** [comment]
I assume you haven't change the "Limit note width" setting?  If so then this is mostly expected, though I agree that it should probably be able to go smaller than the fixed width, just not  larger.

**Comment by toolstack** [comment]
Ok, so I looked into this and it appears to be expected behaviour with GTK4/Adwaita apps and the SourceView control that Folio uses.

Basically "Limit note width" is mislabelled, it should be "Set fixed note width".

Trying to auto re-size the widget would go against several of the GTK4/Adwaita principals so I'm not going to attempt that as it will just break again in the future more than likely.

Instead I'm going to commit two changes to resolve this issue:
- Relabel the preference
- Add a new pref to allow the user to set the fixed width size, at the moment it is 720, which will be the default

**Comment by alice-mkh** [comment]
libadwaita maintainer here, someone pinged me to take a look at this issue and I have no idea what principles you mean.

1. AdwClampScrollable
2. While looking at it I noticed that there's a toast overlay inside the scrolled window?.. Why? Put it right into the window instead
3. GtkSourceView inside a GtkViewport (or AdwToastOverlay, for that matter) is not supported, so you likely broken scrolling on large notes right now. AdwClampScrollable should fix that too

**Comment by toolstack** [comment]
@alice-mkh thanks, I'm new to GTK/libadwaita but from what I saw it looks like monitoring window size changes is discouraged, which seemed to be the only way to reset the size of the sourceview widget if it was set to a fixed width instead of letting it fill the window.

I'll take a look at AdwClampScrollable.

**Comment by alice-mkh** [comment]
You don't need to monitor window sizes here, no.

> if it was set to a fixed width

Right, don't do this. And you have to get rid of that toast overlay too (and of viewport if you added it manually, clamp scrollable replaces that)

**Comment by toolstack** [comment]
The primary notes area, by default, has a fixed with, so I can't just undo that easily without breaking existing behaviour.  As such I've implemented an alternate solution that works for Folio.

I'll probably look at changing the default to non-fixed width in a future release though.

I took a quick look at clamp scrollable and don't think it will fit within the app at the moment, but I do have to do a major change at some point as Folio currently uses a lot of deprecated widgets, so I'll look more at it when I get to that.
